Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs

```diff
@@ -34,28 +37,56 @@ public class RipcordBehavior : MonoBehaviour
     private int _numReleased;
     private GameObject _preview;
 
+
+
+    private bool _gameStarted;
+
+    [Header("Light")]
+    private int _currentLight;
+    [SerializeField] private GameObject[] _listLights;
+    private float _lightSpeed;
+    private bool _lightDoOnce;
+
     [SerializeField] private GameObject _gears;
 
     private PlayerController _pc;
 
     // Start is called before the first frame update
     void Start()
     {
+        _switchMove = -1;
         PressedE = false;
         _doOnce = true;
         _doOnce2 = true;
         _hasReleased = false;
         _numReleased = 0;
         _relocatePoint = new Vector3(transform.position.x, transform.position.y, transform.position.z);
-
+        _gameStarted = false;
         GameObject _pcObject = GameObject.FindWithTag("Player");
         _pc = _pcObject.GetComponent<PlayerController>();
+        _currentLight = 0;
+        _lightDoOnce = true;
+        _switchDoOnce = true;
     }
     // Update is called once per frame
     void Update()
     {
         _successfulPulls.text = _numReleased.ToString();
-        if (_pc.Interact.IsPressed() && PressedE == true && transform.position.z > _targetFollow.transform.position.z + 1.7f && _numReleased != 3)
+        if(_pc.Interact.IsPressed() && _switchDoOnce == true && _inSwitchPos == true)
```

Comment:
SwitchDoOnce can be renamed, you also dont need to say `== true`, `if(variable)` is the same thing, just easier to read and cleaner